crypto/asn1/a_time.c

```diff
@@ -200,16 +393,8 @@ int ASN1_TIME_to_tm(const ASN1_TIME *s, struct tm *tm)
         return 0;
     }
 
-    if (s->type == V_ASN1_UTCTIME) {
-        memset(tm, 0, sizeof(*tm));
-        return asn1_utctime_to_tm(tm, s);
-    }
-    if (s->type == V_ASN1_GENERALIZEDTIME) {
-        memset(tm, 0, sizeof(*tm));
-        return asn1_generalizedtime_to_tm(tm, s);
-    }
-
-    return 0;
+    memset(tm, 0, sizeof(*tm));
```

Comment:
If you examine what are the remaining fields of the struct tm, i.e. those *not* listed, you'll see that they are tm_wday, tm_yday and tm_isdst. Confusion arose from the assumption that that there are more fields, or rather such that would be considered as worth preserving. Once again, wording "only fields x, y, z, are updated" kind of implies that caller is *entitled* to expect that other fields are preserved and worth preserving. But after closer look it's obvious that it's essentially unnatural to expect this. It's only natural that non-listed fields are zeroed. And that's what happens, and even without referred memset, asn1_time_to_tm does it in either case.

To summarize, bottom line is that it's sufficient to omit memset, and in additoin to that I'd argue in favour of rewording documentation to avoid the misconception.